rows.go

```diff
@@ -197,6 +213,14 @@ func (rows *snowflakeRows) NextResultSet() error {
 	return rows.ChunkDownloader.nextResultSet()
 }
 
+func (rows *snowflakeRows) checkAsyncQueryStatus() {
```

Comment:
I would call this something like `waitForQueryStatusIfNeeded()` to make it clear at the call site that this function might block.